extensions/wikia/SitemapXml/SitemapHooks.class.php

```diff
@@ -6,10 +6,72 @@ class SitemapHooks {
 	 * Disable title redirects for sitemaps, as we want to use the original request url rather than Special:Sitemap.
 	 */
 	public static function onBeforeTitleRedirect( WebRequest $request, Title $title ): bool {
-		if ( $title->isSpecial( 'Sitemap' ) ) {
+		if ( self::isSitemapTitle( $title ) ) {
 			return false;
 		}
 		return true;
 	}
 
+	public static function onTestCanonicalRedirect( WebRequest $request, Title $title, OutputPage $output ) {
+		global $wgServer, $wgScriptPath, $wgScript, $wgArticlePath;
+
+		if ( !self::isSitemapTitle( $title ) || WebRequest::detectProtocol() !== 'http' ) {
+			return true;
+		}
+
+		$currentHost = parse_url( WikiFactoryLoader::getCurrentRequestUri( $_SERVER, true, true ), PHP_URL_HOST );
+
+		// We only want to deal with plain HTTP versions of the wikia.com domain
+		if ( !self::shouldCancelRedirect( $currentHost, $title ) ) {
+			return true;
+		}
+
+		$output->cancelRedirect();
+
+		$request->response()->header( 'X-sitemaps-redirect-cancelled: 1' );
```

Comment:
I think the convention is to use capitalized header names, i.e. `X-Sitemaps-Redirect-Cancelled`

Reply:
:+1: Done.